WordPress/gutenberg
# External link component: Add a check for on page anchor links

## What?

This is an attempt to solve an issue where clicking an anchor link inside the link control modal opens the editor in a new tab.
Please see #10117.

This PR prevents following the anchor link by checking if the `href` starts with # and then using `event.preventDefault` on click. There is also a valid suggestion in the issue, to open the link but with a front view, not editor view. I was unsure of how to make that change, so I went with this smaller change.

While I believe that this PR works, it feels like treating the symptoms rather than the cause.
Does anyone have insight in why the external link component is not only used for, well, **external** links?

## Testing Instructions

1. Open the editor and create a new post or page.
2. Next add a block with a link control like a paragraph block or button block. Add an anchor link (`#test`).
3. Click the link in the link control modal. Nothing should happen.
4. Please also test an internal content link (I mean a link to a post or page on the same install) and an external link to see that they still open correctly.

## Screenshots or screencast
